internal/copa/cli.go

```diff
@@ -0,0 +1,301 @@
+	Error                   string
+	Duration                time.Duration
+	VexPath                 string // Only populated for report-based patching
+	UpdatedPackageCount     int
+	FixedVulnerabilityCount int
+}
+
+// CopaSupportedPlatforms lists all platforms that Copa can patch
+// Based on Copa documentation: https://project-copacetic.github.io/copacetic/website/multiplatform-patching
+var CopaSupportedPlatforms = []string{
+	"linux/amd64",
+	"linux/arm64",
+	"linux/arm/v7",
+	"linux/arm/v6",
+	"linux/386",
+	"linux/ppc64le",
+	"linux/s390x",
+	"linux/riscv64",
+}
+
+type CLI struct {
+	copaPath   string
+	dryRun     bool
+	image      string
+	tag        string
+	platforms  []string
+	push       bool
+	reportPath string
+	vexPath    string
+	cmd        *exec.Cmd   // Current command being built
+	dockerAuth docker.Auth // Dependency injection for docker authentication
+}
+
+type PatchParamsConstraint interface {
+	types.ReportBasedPatchParams |
+		types.PlatformSelectivePatchParams |
+		types.ComprehensivePatchParams
+}
+
+// NOTE: use generic for param types to assist the agent with populating the correct values.
+func New[T PatchParamsConstraint](params T, dryRun bool) *CLI {
+	var image, tag, reportPath string
+	var platforms []string
+	var push bool
+
+	// Extract common fields using type switch
+	switch p := any(params).(type) {
+	case types.ReportBasedPatchParams:
+		image, tag, push, reportPath = p.Image, p.Tag, p.Push, p.ReportPath
+	case types.PlatformSelectivePatchParams:
+		image, tag, push, platforms = p.Image, p.Tag, p.Push, p.Platform
+	case types.ComprehensivePatchParams:
+		image, tag, push = p.Image, p.Tag, p.Push
+	}
+
+	return &CLI{
+		copaPath:   "copa",
+		dryRun:     dryRun,
+		image:      image,
+		tag:        tag,
+		platforms:  platforms,
+		push:       push,
+		reportPath: reportPath,
+		dockerAuth: &docker.AuthImpl{}, // Default to real implementation
+	}
+}
+
+// NewWithDockerAuth creates a CLI instance with custom docker auth for testing
+func NewWithDockerAuth[T PatchParamsConstraint](params T, dryRun bool, dockerAuth docker.Auth) *CLI {
+	cli := New(params, dryRun)
+	cli.dockerAuth = dockerAuth
+	return cli
+}
+
+func (c *CLI) Build() *CLI {
+	args := []string{"patch"}
+	args = append(args, "--image", c.image)
+
+	if c.tag != "" {
+		args = append(args, "--tag", c.tag)
+	}
+
+	if c.push {
+		args = append(args, "--push")
+	}
+
+	c.cmd = exec.Command(c.copaPath, args...)
+	return c
+}
+
+func (c *CLI) BuildWithPlatforms() *CLI {
+	c = c.Build()
+
+	if len(c.platforms) > 0 {
+		supportedPlatforms := FilterSupportedPlatforms(c.platforms)
+		if len(supportedPlatforms) > 0 {
+			c.cmd.Args = append(c.cmd.Args, "--platform", strings.Join(supportedPlatforms, ","))
+		}
+	}
+
+	return c
+}
+
+func (c *CLI) BuildWithReport() *CLI {
+	c = c.Build()
+
+	if c.reportPath != "" {
+		c.cmd.Args = append(c.cmd.Args, "--report", c.reportPath)
+		c.vexPath = filepath.Join(os.TempDir(), defaultVexFile)
```

Comment:
I think `TempDir()` points to a single static folder, could this result in multiple mcp-server containers reading/writing to the same vex document? If so, perhaps we can create a new dir for it with `os.MkdirTemp()`?

Reply:
Good call, I'll make that change. :) Thanks for the suggestion!